tutorials/intermediate/blitting.py

```diff
@@ -0,0 +1,183 @@
+"""
+=================
+Blitting Tutorial
+=================
+
+'Blitting' is a `standard technique
+<https://en.wikipedia.org/wiki/Bit_blit>`__ in computer graphics that
+in the context of matplotlib can be used to (drastically) improve
+performance of interactive figures.  It is used internally by the
+:mod:`~.animation` and :mod:`~.widgets` modules for this reason.
+
+The source of the performance gains is simply not re-doing work we do
+not have to.  For example, if the limits of an Axes have not changed,
+then there is no reason we should re-draw all of the ticks and
+tick-labels (particularly because text is one of the more expensive
+things to render).
+
+The procedure to save our work is roughly:
+
+- draw the figure, but exclude an artists marked as 'animated'
+- save a copy of the Agg RBGA buffer
+
+In the future, to update the 'animated' artists we
+
+- restore our copy of the RGBA buffer
+- redraw only the animated artists
+- show the resulting image on the screen
+
+thus saving us from having to re-draw everything which is _not_
+animated.
+
+Simple Example
+--------------
+
+We can implement this via methods on `.CanvasAgg` and setting
+``animated=True`` on our artist.
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+x = np.linspace(0, 2*np.pi, 100)
+
+fig, ax = plt.subplots()
+# animated=True makes the artist be excluded from normal draw tree
+ln, = ax.plot(x, np.sin(x), animated=True)
+
+# stop to admire our empty window axes and ensure it is drawn
+plt.pause(.1)
+
+# save a copy of the image sans animated artist
+bg = fig.canvas.copy_from_bbox(fig.bbox)
+# draw the animated artist
+ax.draw_artist(ln)
+# show the result to the screen
+fig.canvas.blit(fig.bbox)
+
+for j in range(100):
+    # put the un-changed background back
+    fig.canvas.restore_region(bg)
+    # update the artist.
+    ln.set_ydata(np.sin(x + (j / 100) * np.pi))
+    # re-render the artist
+    ax.draw_artist(ln)
+    # copy the result to the screen
+    fig.canvas.blit(fig.bbox)
+
```

Comment:
OK, I ran this using `Qt5Agg` and an axis pops up with the default axis limits, but nothing is drawn 😢 If I run on MacOSX nothing is popped up.  The loop runs, as tested via a print statement...

Reply:
If you add `plt.ion` to the top does it work? 

Adding `plt.pause(.1)` in here may also help...

Reply:
I did:

```python
x = np.linspace(0, 2*np.pi, 100)

fig, ax = plt.subplots()

plt.ion()
# animated=True makes the artist be excluded from normal draw tree
ln, = ax.plot(x, np.sin(x), animated=True)
# stop to admire our empty window axes and ensure it is drawn
plt.pause(.1)
```

it still doesn't work.

Reply:
Bump the iteration from 100 to 1000. It runs to fast that you almost do not see anything.
A pause in the loop might help. I found that plt.show() at the end do not destroy the figure when the loop is finished.

Is there another way to do this without using the plt.pause() function? The function doc says it is an experimental function.

Reply:
No window update at all happens, and the original artist is never drawn; just an empty window: Qt5Agg, MacOSx, TkAgg, on master, MacOSX 10.13.2.  The loop certainly runs, and terminates normally, but the plot is never drawn or animated...

Reply:
I believe there could be a problem with copying the background. For example the window get resized after the first copying. 
Though you should still be able to see the plot.
Have you tried the method in the example with dynamic update of background?

Reply:
Oh, hmmm.  I'm on a hidpi monitor (retina).  Is that what you mean?

Reply:
This works for me if I add `plt.ion()` at the top, using Qt5Agg.

Reply:
If I do `python blitting.py` as-is I see animation with both tkagg and qt5agg on master and 3.2.1.

Reply:
Just chiming in that this works on my computer as-is.